### PR TITLE
VitePress-powered documentation site

### LINKED ISSUE #1443: Tracking issue for new documentation
Ref. #1435
- [x] [index.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/index.md)
- [x] [what-is-plot.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/what-is-plot.md)
- [x] [why-plot.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/why-plot.md)
- [x] [getting-started.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/getting-started.md)
- [x] [community.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/community.md)
- [x] [features/plots.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/plots.md)
- [x] [features/marks.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/marks.md)
- [x] [features/scales.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/scales.md)
- [x] [features/projections.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/projections.md)
- [x] [features/transforms.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/transforms.md)
- [x] [features/facets.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/facets.md)
- [x] [features/legends.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/legends.md)
- [x] [features/curves.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/curves.md)
- [x] [features/formats.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/formats.md)
- [x] [features/markers.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/markers.md)
- [x] [features/shorthand.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/shorthand.md)
- [x] [features/spatial-interpolators.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/spatial-interpolators.md)
- [x] [features/accessibility.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/features/accessibility.md)
- [x] [marks/area.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/area.md)
- [x] [marks/arrow.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/arrow.md)
- [x] [marks/auto.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/auto.md)
- [x] [marks/axis.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/axis.md)
- [x] [marks/bar.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/bar.md)
- [x] [marks/box.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/box.md)
- [x] [marks/cell.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/cell.md)
- [x] [marks/contour.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/contour.md)
- [x] [marks/delaunay.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/delaunay.md)
- [x] [marks/density.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/density.md)
- [x] [marks/dot.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/dot.md)
- [x] [marks/frame.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/frame.md)
- [x] [marks/geo.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/geo.md)
- [x] [marks/grid.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/grid.md)
- [x] [marks/hexgrid.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/hexgrid.md)
- [x] [marks/image.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/image.md)
- [x] [marks/line.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/line.md)
- [x] [marks/linear-regression.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/linear-regression.md)
- [x] [marks/link.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/link.md)
- [x] [marks/raster.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/raster.md)
- [x] [marks/rect.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/rect.md)
- [x] [marks/rule.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/rule.md)
- [x] [marks/text.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/text.md)
- [x] [marks/tick.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/tick.md)
- [x] [marks/tree.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/tree.md)
- [x] [marks/vector.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/marks/vector.md)
- [x] [transforms/bin.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/bin.md)
- [x] [transforms/centroid.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/centroid.md)
- [x] [transforms/dodge.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/dodge.md)
- [x] [transforms/filter.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/filter.md)
- [x] [transforms/group.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/group.md)
- [x] [transforms/hexbin.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/hexbin.md)
- [x] [transforms/interval.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/interval.md)
- [x] [transforms/map.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/map.md)
- [x] [transforms/normalize.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/normalize.md)
- [x] [transforms/select.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/select.md)
- [x] [transforms/sort.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/sort.md)
- [x] [transforms/stack.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/stack.md)
- [x] [transforms/tree.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/tree.md)
- [x] [transforms/window.md](https://github.com/observablehq/plot/blob/mbostock/vitepress/docs/transforms/window.md)

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #1443.

### CODE REVIEW

**Comment by Fil** [comment]
`:::plot` is a _beautiful_ way to address the requirements. It makes authoring a very pleasant experience, with proper color highlighting and clutter-free code.

I'm not sure if there is a legal way of adding names or comments next to :::plot, but it could make it even nicer with the project search:
vscode-plot

**Comment by Fil** [comment]
I've added `:::plot` on all of scales.md and hit two problems:

1. interactive variables, such as _padding_ and _align_ which are a viewof in the original notebook.
2. datasets (e.g., gistemp). Maybe with the *data* pattern defined in https://vitepress.dev/guide/data-loading?

File: docs/marks/dot.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+```js
+Plot.plot({
+  x: {
+    label: null,
+    tickSize: 0
+  },
+  y: {
+    label: "↑ Frequency (%)",
+    transform: (d) => d * 100
```
Comment (Fil) [inline docs/marks/dot.md:170]:
```suggestion
    percent: true
```

File: docs/features/marks.md [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+```js
+Plot.barX({length: 7}, {
+  x: [8, 6, 7, 5, 3, 0, 9],
+  y: [0, 1, 2, 3, 4, 5, 6]
+}).plot()
+```
+:::
+
+:::info
+This example uses parallel arrays, an alternative method of specifying [channels](#marks-have-channels). There are seven (7) bars in total. The first element in each array gives the **x** and **y** values for the first bar (4 and 0); the second element gives the values for the second bar (3 and 1); and so on. This is convenient here for supplying hard-coded values, but more commonly channel values are derived from data.
```
Comment (Fil) [inline docs/features/marks.md:326]:
```suggestion
This example uses parallel arrays, an alternative method of specifying [channels](#marks-have-channels). There are seven (7) bars in total. The first element in each array gives the **x** and **y** values for the first bar (8 and 0); the second element gives the values for the second bar (6 and 1); and so on. This is convenient here for supplying hard-coded values, but more commonly channel values are derived from data.
```

File: docs/features/curves.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import {ref} from "vue";
+
+const curve = ref("catmull-rom");
+const numbers = d3.range(20).map(d3.randomLcg(42));
+
+</script>
+
+# Curves
+
+A curve defines how to turn a discrete representation of a line as a sequence of points [[*x₀*, *y₀*], [*x₁*, *y₁*], [*x₂*, *y₂*], …] into a continuous path; *i.e.*, how to interpolate between points. Curves are used by the [line](../marks/line.md), [area](../marks/area.md), and [link](../marks/link.md) mark, and are implemented by [d3-shape](https://github.com/d3/d3-shape/blob/main/README.md#curves).
```
Comment (Fil) [inline docs/features/curves.md:14]:
```suggestion
A curve defines how to turn a discrete representation of a line as a sequence of points [[*x₀*, *y₀*], [*x₁*, *y₁*], [*x₂*, *y₂*], …] into a continuous path; *i.e.*, how to interpolate between points. Curves are used by the [line](../marks/line.md), [area](../marks/area.md), and [link](../marks/link.md) marks, and are implemented by [d3-shape](https://github.com/d3/d3-shape/blob/main/README.md#curves).
```

File: docs/features/shorthand.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+:::
+
+And here’s the [dodge transform](../transforms/dodge.md) for a beeswarm plot:
+
+:::plot
+```js
+Plot.dotX(numbers, Plot.dodgeY()).plot()
+```
+:::
+
```
Comment (Fil) [inline docs/features/shorthand.md:208]:
```suggestion

## Tuples

```

(maybe with a "Lists" subheader at the top?)

File: docs/marks/line.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+  marks: [
+    Plot.line(sftemp, Plot.windowY(14, {x: "date", y: "low", stroke: "#4e79a7"})),
+    Plot.line(sftemp, Plot.windowY(14, {x: "date", y: "high", stroke: "#e15759"})),
+    Plot.ruleY([32]) // freezing
+  ]
+})
+```
+:::
+
+If some channel values are undefined (or null or NaN), gaps will appear between adjacent points. To demonstrate, below we set the **y** value to NaN for the first three months of each year.
```
Comment (Fil) [inline docs/marks/line.md:245]:
```suggestion
If some channel values are undefined (or null, NaN, or infinite), gaps will appear between adjacent points. To demonstrate, below we set the **y** value to NaN for the first three months of each year.
```

File: docs/marks/link.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  x: {
+    type: "log",
+    label: "Population →"
+  },
+  y: {
+    label: "↑ Inequality",
+    ticks: 4
+  },
+  color: {
+    scheme: "BuRd",
```
Comment (Fil) [inline docs/marks/link.md:41]:
```suggestion
    scheme: "BuRd",
    symmetric: false,
```

without the symmetric option, we don't see any blue link.

File: docs/features/shorthand.md (hunk 4 above)
Comment (mbostock) [reply docs/features/shorthand.md:210]: Added “One dimension”, “Two dimensions”, and “Caveats” as sections.

File: docs/marks/link.md (hunk 6 above)
Comment (mbostock) [reply docs/marks/link.md:41]: I tried that, but I don’t like the way the legend looks. I think they are still technically “blue” even though they are a very very pale blue that is almost indistinguishable from white.

Screenshot 2023-04-17 at 3 43 23 PM

File: docs/marks/rect.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+The following channels are optional:
+
+* **x1** - the starting horizontal position; bound to the *x* scale
+* **y1** - the starting vertical position; bound to the *y* scale
+* **x2** - the ending horizontal position; bound to the *x* scale
+* **y2** - the ending vertical position; bound to the *y* scale
+
+Typically either **x1** and **x2** are specified, or **y1** and **y2**, or both.
+
+If an **interval** is specified, such as d3.utcDay, **x1** and **x2** can be derived from **x**: *interval*.floor(*x*) is invoked for each **x** to produce **x1**, and *interval*.offset(*x1*) is invoked for each **x1** to produce **x2**. The same is true for **y**, **y1**, and **y2**, respectively. If the interval is specified as a number *n*, **x1** and **x2** are taken as the two consecutive multiples of *n* that bracket **x**.
```
Comment (Fil) [inline docs/marks/rect.md:203]:
```suggestion
If an **interval** is specified, such as d3.utcDay, **x1** and **x2** can be derived from **x**: *interval*.floor(*x*) is invoked for each **x** to produce **x1**, and *interval*.offset(*x1*) is invoked for each **x1** to produce **x2**. The same is true for **y**, **y1**, and **y2**, respectively. If the interval is specified as a number *n*, **x1** and **x2** are taken as the two consecutive multiples of *n* that bracket **x**. Time intervals can also be specified by name (_e.g._ _quarter_ for d3.utcMonth.every(3)).
```

**Review by Fil — APPROVED** [review]

**Pull request merged**